ministryofjustice/laa-assess-crime-forms
# CRM457-1949 delete adjustments

## Description of change

delete single and all adjustments made by CW for CRM7 

[Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-1949)

## Notes for reviewer
- an event `Event::UndoEdit` is created for each and every `delete` of an adjustment
- for a 'delete all adjustments' interaction each 'deletion' of an event records an `Event::Undo` and one single `Event::DeleteAdjustments` that is shown in claim history
- a 'deleted' adjustment is a `work_type`, `letter_and_call` or `disbursement` that is **reverted** to its original values
- `Services/AdjustmentDeleter` is now a base class that `PA` and `NSM` `AdjustmentDeleter` services inherit from
- redirects after a delete got to the type adjustment page listing adjustments unless there are no adjustments of any type and redirect will go to `claims/work_item`#index` page
## Screenshots of changes (if applicable)

### Before changes:
A CW is unable to delete adjustment changes

### After changes:
A CW can delete any adjustment they have made and costs/quantities will revert to original values

## How to manually test the feature

1. Log-in as CW and create some adjustments for existing work_items, calls_and_letters, or disbursements
2. Delete single adjustments 
3. Adjustment should no longer show and item should have reverted to original value

1. Log-in as CW and create some adjustments for existing work_items, calls_and_letters, or disbursements
2. Go to adjustments and click delete all adjustments
4. Check adjustments do not exist and items have revereted to original values
5. check that a delete all adjustments event is created with a comment in the claims history section

https://crm457-1949-delete-nscc-caseworker-dev.cloud-platform.service.justice.gov.uk/
